### ISSUE TITLE
[iOS][Accessibility] VoiceOver i18n accessibilityState read in English only

### ISSUE THREAD

**Issue by draperunner** [facebook/react-native#28328]
When using some accessibilityStates, the state postfix/prefix is read out in English, not the phone's language on iOS.

It seems like `accessibilityState={{ selected }}` translates correctly, but `checked` and `collapsed` do not. I have not tried the others.

React Native version: 0.61.5

```
System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
    Memory: 101.02 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.13.0 - ~/.nvm/versions/node/v10.13.0/bin/node
    Yarn: 1.22.4 - ~/.yvm/shim/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v10.13.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 23, 25, 26, 27, 28
      Build Tools: 23.0.1, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.0, 27.0.3, 28.0.2, 28.0.3
      System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom_64, android-25 | Google APIs Intel x86 Atom_64, android-25 | Google Play Intel x86 Atom, android-26 | Google APIs Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom
      Android NDK: 17.1.4828580
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5791312
    Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0
    react-native: 0.61.5 => 0.61.5
  npmGlobalPackages:
    react-native-cli: 2.0.1
    react-native-git-upgrade: 0.2.7
    react-native-permissions: 2.0.1
    react-native: 0.61.5
```

## Steps To Reproduce
1. Create a clean project: `npx react-native init a11ybug`
2. Run it on an iPhone device with phone language not English (Norwegian in my case)
3. Add the following to your App.js:

```
        <TouchableOpacity
          accessibilityRole="checkbox"
          accessibilityState={{ checked: true }}>
          <View>
            <Text>Trykk her</Text>
          </View>
        </TouchableOpacity>
```
4. Make sure to activate VocieOverx

When you mark the checkbox above, VoiceOver will read "Trykk her checkbox checked".

## Expected Results

I am not sure what the exact correct translation would be for this, but I'd expect something like "Trykk her, markert avmerkingsknapp."

**Comment by chrisglein** [comment]
This is not implemented yet. There's some design work to figure out how to not bundle all the languages in React Native core so there's not huge bloat here.

**Comment by draperunner** [comment]
I see. Is there a public discussion regarding this somewhere?

I'd be happy to help out with Norwegian translations if/when it comes to that. :)

**Comment by chrisglein** [comment]
> I see. Is there a public discussion regarding this somewhere?

Not to my knowledge (although you could check the [discussions](https://github.com/react-native-community/discussions-and-proposals/issues/) repo). Mostly I think this is functionality that hasn't been filled in yet.

**Comment by TheSavior** [comment]
I think we can keep the conversation here instead of the discussions repo.

My understanding is that the reason we don't have this is because including translations for all the accessibility options in all languages would make React Native massive.

I think a solution to this problem would require solving that problem. I'm not familiar enough with the space to have any recommendations, but that has been the blocker historically.

**Comment by draperunner** [comment]
Okay, so I'd like to come with some thoughts to get the discussion started. Note that I have very little knowledge about the details of screen readers and how it's implemented in React Native.

I agree that keeping translations outside the core is a good idea to reduce bundle size. So, creating a separate package comes to mind. But I doubt many app developers care about including _all_ existing languages, so we need a way of picking just those that your app supports. I'm thinking maybe providing a scoped package for each language might be a valid solution. So you you would install something like `@react-native-a11y-i18n/en`, `@react-native-a11y-i18n/de`, `@react-native-a11y-i18n/no`, or something like that. Maybe a `@react-native-a11y-i18n/all` if you really need all of them.

And then we need an API for applying these translations. There are different strategies we can go for here. I'd like to mention that in our app we let the user select their preferred language independently from the device language.

## An Imperative API
We could create a new method on the Accessibility API called `setLocale` or something that loads a locale from the above-mentioned scoped packages:

```
import { Accessibility } from 'react-native'

import german from '@react-native-a11y-i18n/de'

Accessibility.setLocale(german)
```

It's easy to create a hook that listens to some global state or the device language. I just learned about [react-native-localize](https://github.com/react-native-community/react-native-localize) which you can use to get the user's preferred languages.

```
import en from '@react-native-a11y-i18n/en '
import de from '@react-native-a11y-i18n/de'

cont LOCALES = { 
  en: en,
  de: de 
}

const App = () => {
  const language = useContext(MyLanguageContext)

  useEffect(() => {
    Accessibility.setLocale(LOCALES[language])
  }, [language])
}
```

I think this API is both intuitive and flexible.

## A Declarative API
Another approach would be to send the locale information to every component as a prop. It would be up to the developer to compose reusable components.

```
import german from '@react-native-a11y-i18n/de'

const Button = ({ children }) => (
  <Touchable
    accessibilityRole="button"
    accessibilityLocale={german}
  >
    { children }
 </Touchable>
)
```

I don't like this approach. It is very prone to errors due to how easy it is to forget to add the locales in every component. It _does_ support using different languages on different components simultaneously – which the imperative API does not – but I cannot imagine that's a common use case...

**Comment by TheSavior** [comment]
I appreciate the detailed proposal!

I think with React Native, the question we often want to start by asking is how do the native platforms handle this? Is there some way we can piggyback on that instead of reimplementing things in user land?

Understanding how native apps make the talkback responses localized would probably be a good starting point.
